src/Common/src/CoreLib/Interop/Windows/Kernel32/Interop.EventWaitHandle.cs

```diff
@@ -13,16 +14,19 @@ internal static partial class Kernel32
         internal const uint CREATE_EVENT_INITIAL_SET = 0x2;
         internal const uint CREATE_EVENT_MANUAL_RESET = 0x1;
 
-        [DllImport(Interop.Libraries.Kernel32, SetLastError = true)]
+        [DllImport(Libraries.Kernel32, SetLastError = true)]
         internal static extern bool SetEvent(SafeWaitHandle handle);
 
-        [DllImport(Interop.Libraries.Kernel32, SetLastError = true)]
+        [DllImport(Libraries.Kernel32, SetLastError = true)]
         internal static extern bool ResetEvent(SafeWaitHandle handle);
 
-        [DllImport(Interop.Libraries.Kernel32, EntryPoint = "CreateEventExW", SetLastError = true, CharSet = CharSet.Unicode)]
+        [DllImport(Libraries.Kernel32, EntryPoint = "CreateEventExW", SetLastError = true, CharSet = CharSet.Unicode)]
         internal static extern SafeWaitHandle CreateEventEx(IntPtr lpSecurityAttributes, string? name, uint flags, uint desiredAccess);
 
-        [DllImport(Interop.Libraries.Kernel32, EntryPoint = "OpenEventW", SetLastError = true, CharSet = CharSet.Unicode)]
+        [DllImport(Libraries.Kernel32, EntryPoint = "OpenEventW", SetLastError = true, CharSet = CharSet.Unicode)]
         internal static extern SafeWaitHandle OpenEvent(uint desiredAccess, bool inheritHandle, string name);
+
+        [DllImport(Libraries.Kernel32, EntryPoint = "CreateEventExW", SetLastError = true, CharSet = CharSet.Unicode, BestFitMapping = false)]
```

Comment:
From our conversation earlier, @JeremyKuhne, instead of using CreateEventW, I should use CraeteEventExW, so I could pass EVENT_ALL_ACCESS to desiredAccess. The problem is that we already have a PInvoke that calls it, but the first argument is an IntPtr. I didn't want to touch that version so I added another one that accepts a ref. Suggestions about the name are welcome. Is this the correct thing to do?

Reply:
Don't create another Import as that is unnecessary overhead (additional IL generation for the new import). You can just create a wrapper here that takes the address of the `SECURITY_ATTRIBUTES` and casts to `IntPtr` for convenience.

Reply:
@JeremyKuhne 

Both of these PInvokes are hitting the same failure of the object disposed exception I mentioned in my other comment:

```cs
[DllImport(Libraries.Kernel32, EntryPoint = "CreateEventW", SetLastError = true, CharSet = CharSet.Unicode, BestFitMapping = false)]
internal static extern SafeWaitHandle CreateEvent(ref SECURITY_ATTRIBUTES lpSecurityAttributes, bool isManualReset, bool initialState, string? name);
```

```cs
[DllImport(Libraries.Kernel32, EntryPoint = "CreateEventExW", SetLastError = true, CharSet = CharSet.Unicode, BestFitMapping = false)]
 internal static extern SafeWaitHandle CreateEvent(ref SECURITY_ATTRIBUTES lpSecurityAttributes, string? name, uint flags, uint desiredAccess);
```
This second one I was calling it by passing:

```cs
            uint eventFlags = initialState ? Interop.Kernel32.CREATE_EVENT_INITIAL_SET : 0;

            if (mode == EventResetMode.ManualReset)
            {
                EventResetMode.ManualReset => true,
                EventResetMode.AutoReset => false,
                _ => throw new ArgumentOutOfRangeException(nameof(mode), SR.Format(SR.ArgumentOutOfRange_Enum))
            };
                eventFlags |= (uint)Interop.Kernel32.CREATE_EVENT_MANUAL_RESET;
            }
            ...
            using SafeWaitHandle handle = Interop.Kernel32.CreateEvent(ref secAttrs, name, eventFlags, (uint)EventWaitHandleRights.FullControl);

```

Reply:
Nevermind! Stephen pointed out that I should not be using the using keyword anymore.

Should I still switch to use the second PInvoke?